Service/ZGWToXxllncZaakService.php

```diff
@@ -0,0 +1,499 @@
+    //     $objectEntity = $this->entityManager->find(ObjectEntity::class, $data['response']['_self']['id']);
+    //     if (!in_array($objectEntity->getEntity()->getName(), ['ZaakEigenschap'])) {
+    //         return $data;
+    //     }
+    //     // var_dump('mapUpdate triggered');
+    //     $this->data = $data['response'];
+    //     $this->configuration = $configuration;
+
+    //     $this->data = $objectEntity->toArray();
+
+    //     isset($this->configuration['entities']['XxllncZaakPost']) && $xxllncZaakPostEntity = $this->entityRepo->find($this->configuration['entities']['XxllncZaakPost']);
+
+    //     if (!isset($xxllncZaakPostEntity)) {
+    //         throw new \Exception('XxllncZaakPost entity not found, check MapUpdateZaak config');
+    //     }
+
+    //     if (!isset($this->data['zaak'])) {
+    //         throw new \Exception('Zaak not set on given object');
+    //     }
+
+    //     if (isset($this->data['zaak']['zaaktype'])) {
+    //         isset($this->data['zaak']['zaaktype']['_self']['id']) && $zaakTypeId = $this->data['zaak']['zaaktype']['_self']['id'];
+    //         $zaakTypeObject = $this->entityManager->find('App:ObjectEntity', $zaakTypeId);
+    //         $casetypeId = $zaakTypeObject->getExternalId();
+    //         // Return here cause if the zaaktype is created through this gateway, we cant sync it to xxllnc because it doesn't exist there
+    //         if (!isset($casetypeId)) {
+    //             return $this->data;
+    //         }
+    //     } else {
+    //         throw new \Exception('ZaakType not set on Zaak');
+    //     }
+
+    //     // @TODO update SyncUpdateZaak config with Zaak externalId?
+    //     $values = $this->entityManager->getRepository('App:Value')->findBy(['stringValue' => $this->data['zaak']['_self']['id']]);
+    //     foreach ($values as $value) {
+    //         if ($value->getObjectEntity()->getEntity()->getId()->toString() == $xxllncZaakPostEntity->getId()->toString() && $value->getAttribute()->getName() == 'zgwZaak') {
+    //             $externalId = $value->getObjectEntity()->getExternalId();
+    //             $zaakArray = $value->getObjectEntity()->toArray();
+    //             break;
+    //         }
+    //     }
+
+    //     if (!isset($externalId)) {
+    //         throw new \Exception('Earlier external id from xxllnc not found');
+    //     }
+
+    //     $zaakObject = $this->entityManager->find('App:ObjectEntity', $this->data['zaak']['_self']['id']);
+    //     if (!$zaakObject instanceof ObjectEntity) {
+    //         throw new \Exception('Zaak object not found with id:' . $this->data['zaak']['_self']['id']);
+    //     }
+
+    //     $zaakArray = $zaakObject->toArray();
+
+    //     $xxllncZaakArray = $this->mapZGWToXxllncZaak($casetypeId, $zaakTypeObject, $zaakArray, $xxllncZaakPostEntity, false);
+
+    //     return ['response' => $xxllncZaakArray, 'entity' => $xxllncZaakPostEntity->getId()->toString()];
+    // }
```

Comment:
Lot of commented out code here, why is this still here? remove or add todo

Reply:
![image](https://user-images.githubusercontent.com/57346398/224293224-7afb8321-80d9-4c8e-91f0-2d407c54b51d.png)
git loopt achter ?